internal/versionsapi/cliinfo.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright (c) Edgeless Systems GmbH
+
+SPDX-License-Identifier: AGPL-3.0-only
+*/
+
+package versionsapi
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"path"
+
+	"github.com/edgelesssys/constellation/v2/internal/constants"
+	"go.uber.org/multierr"
+	"golang.org/x/mod/semver"
+)
+
+// CLIInfo contains information about a specific CLI version (i.e. it's compatibility with Kubernetes versions).
+type CLIInfo struct {
+	// Ref is the reference name of the image.
+	Ref string `json:"ref,omitempty"`
+	// Stream is the stream name of the image.
+	Stream string `json:"stream,omitempty"`
+	// Version is the version of the image.
+	Version string `json:"version,omitempty"`
+	// Kubernetes contains all compatible Kubernetes versions.
+	Kubernetes []string `json:"kubernetes,omitempty"`
+}
+
+// JSONPath returns the S3 JSON path for this object.
+func (c CLIInfo) JSONPath() string {
+	return path.Join(
+		constants.CDNAPIPrefix,
+		"ref", c.Ref,
+		"stream", c.Stream,
+		c.Version,
+		"cli",
+		"info.json",
+	)
+}
+
+// URL returns the URL to the JSON file for this object.
+func (c CLIInfo) URL() (string, error) {
+	url, err := url.Parse(constants.CDNRepositoryURL)
+	if err != nil {
+		return "", fmt.Errorf("parsing CDN URL: %w", err)
+	}
+	url.Path = c.JSONPath()
+	return url.String(), nil
+}
+
+// ValidateRequest validates the request parameters of the list.
+// The Kubernetes slice must be empty.
+func (c CLIInfo) ValidateRequest() error {
+	var retErr error
+	if err := ValidateRef(c.Ref); err != nil {
+		retErr = multierr.Append(retErr, err)
```

Comment:
```suggestion
        retErr = errors.Join(retErr, err)
```
we no longer use multierr, but the new native way of joining errors